### ISSUE TITLE
SQL Plugins does not fetch SSHKeys

### ISSUE THREAD

**Issue by maikelcoke** [glauth/glauth#258]
Hi,

i use gitea with glauth (postgres plugin). When gitea fetches the user info from glauth (basesqlhandler->FindPosixAccounts), SSHKeys are not supported / implemented. There is no field in the database table and the sql query does not fetch that field, so that sqlbasehandler->getAccount could assign the SSHKeys to the response. Some other Array-Fields are also missing. I tried to fix this for myself, but i failed at the moment. I'm new in go, coming from PHP, Ruby and C#.

If anyone can fix this faster than me, would be nice. Or i will try it in the next days :-)

Have a nice weekend.

**Comment by Fusion** [comment]
Yes, it's an unfortunate limitation at this time: multi-valued fields are not, at this time, properly handled. It's a good short term goal for me to have :)

**Comment by Fusion** [comment]
Possible temporary workaround:

Looking at my code, it appears that I do this thing where, if you also define your user (minimally) in config.cfg, you can specify SSH keys in there, and that will be merged in the returned LDAP structure. Again, not the final fix!

**Comment by Hellhium** [comment]
I saw this getting implemented in #262
Is this ready to be released ?

**Comment by bodsch** [comment]
This does not seem to me to be the optimum DB design.
As soon as more than one ssh-key is to be defined for the user, an extra table - such as with includegroups - would be much more efficient. Actually, this concerns all data that is available in a list ...

**Comment by Fusion** [comment]
This is due to trying to keep the configuration as simple as possible. If a configuration tool could be used to enter the database information, this would be a non-issue. Not sure how to strike a balance between purity and ease of use.

**Comment by bodsch** [comment]
I understand you very well! :)
I'm just trying to look at this from an automation perspective and as a former (rather poor) DB admin.
Maybe you could link up with other projects (e.g. [glauth-ui.light](https://github.com/yvesago/glauth-ui-light))?
Then one could certainly tackle such issues in a more targeted and joint way.
